### DISCUSSION TITLE
must_cast from NoUninit+Zeroable to array?

### DISCUSSION THREAD

**Discussion by James-E-A** [Lokathor/bytemuck#237] [Q&A] [answered]
I've got a struct that's Copy, Clone, repr(C,packed), bytemuck::NoUninit, and bytemuck::Zeroable &mdash; but it is *not* bytemuck::AnyBitPattern due to being slightly value-constrained \(certain fields have certain bits clamped to 0 when creating the type\).

I have an array of this struct, and I need to lend this array *as* a `[u8; _]`.

Is there any plan for a `must_*` version of the `bytes_of` function? I hate inviting run-time land-mines by choosing functions that panic over functions that will fail to compile when I've done something incorrect.

**Comment by zachs18** [comment, marked as answer]
You can use `bytemuck::must_cast_ref`, assuming you have the `min_const_generics` feature enabled, to cast from `&T` (including `&[U; _]`) to `&[u8; N]`

**Reply by James-E-A** [reply]
Ah, this was my bad. I was confused about the type constraints. It does indeed work.

**Comment by Lokathor** [comment]
In addition to what zachs said, casting any type that `bytes_of` can be called with to a slice of bytes already won't ever fail.
